programs/drift/src/controller/orders.rs

```diff
@@ -398,6 +399,147 @@ pub fn place_perp_order(
     Ok(())
 }
 
+pub fn place_and_match_rfq_orders<'c: 'info, 'info>(
+    taker_account_loader: &AccountLoader<'info, User>,
+    taker_stats_account_loader: &AccountLoader<'info, UserStats>,
+    rfq_matches: Vec<RFQMatch>,
+    maker_rfq_account_map: BTreeMap<Pubkey, AccountLoader<'c, RFQUserAccount>>,
+    makers_and_referrer: &UserMap,
+    makers_and_referrer_stats: &UserStatsMap,
+    perp_market_map: &PerpMarketMap,
+    spot_market_map: &SpotMarketMap,
+    oracle_map: &mut OracleMap,
+    state: &State,
+) -> Result<()> {
+    #[cfg(all(feature = "mainnet-beta", not(feature = "anchor-test")))]
+    {
+        panic!("RFQ orders are disabled on mainnet-beta");
+    }
+
+    let taker_key = taker_account_loader.key();
+    let clock = &Clock::get()?;
+
+    for rfq_match in rfq_matches {
+        if rfq_match.maker_order_params.max_ts < clock.unix_timestamp {
+            msg!(
+                "RFQ order expired for maker authority {}",
+                rfq_match.maker_order_params.authority
+            );
+            continue;
+        }
+
+        let mut taker = load_mut!(taker_account_loader)?;
+        let maker_order_params = rfq_match.maker_order_params;
+        let (maker_pubkey, _) = Pubkey::find_program_address(
+            &[
+                &b"user"[..],
+                maker_order_params.authority.as_ref(),
+                &maker_order_params.sub_account_id.to_le_bytes(),
+            ],
+            &ID,
+        );
+
+        let maker_order_params = OrderParams {
+            order_type: OrderType::Limit,
+            market_type: maker_order_params.market_type,
+            market_index: maker_order_params.market_index,
+            direction: maker_order_params.direction,
+            base_asset_amount: rfq_match.base_asset_amount,
+            price: maker_order_params.price,
+            max_ts: Some(maker_order_params.max_ts),
+            immediate_or_cancel: true,
+            post_only: PostOnlyParam::TryPostOnly,
+            ..OrderParams::default()
+        };
+        let mut maker = makers_and_referrer.get_ref_mut(&maker_pubkey)?;
+
+        // See if the UUID already exists in the RFQ Account data
+        let mut rfq_account =
+            load_mut!(maker_rfq_account_map.get(&maker_pubkey).ok_or_else(|| {
+                msg!("RFQ account not found for maker {}", maker_pubkey);
+                ErrorCode::InvalidRFQUserAccount
+            })?)?;
+
+        let rfq_order_id = RFQOrderId::new(
```

Comment:
i think you can just add the Copy derivation to RFQOrderId and then you wont need to do this this manually cloning